classes/migration/upgrade/v3_4_0/jobs/FixRegionCodesDaily.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/**
+ * @file classes/migration/upgrade/v3_4_0/jobs/FixRegionCodesDaily.php
+ *
+ * Copyright (c) 2022-2025 Simon Fraser University
+ * Copyright (c) 2022-2025 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class FixRegionCodesDaily
+ *
+ * @ingroup jobs
+ *
+ * @brief Fixes the wrong region codes, using the temporary table of FIPS-ISO mapping.
+ */
+
+namespace PKP\migration\upgrade\v3_4_0\jobs;
+
+use Illuminate\Support\Facades\DB;
+use PKP\config\Config;
+use PKP\jobs\BaseJob;
+
+class FixRegionCodesDaily extends BaseJob
+{
+    /** The range of metrics_submission_geo_daily_ids to consider for this update */
+    protected int $startId;
+    protected int $endId;
+
+    /**
+     * Create a new job instance.
+     */
+    public function __construct(int $startId, int $endId)
+    {
+        parent::__construct();
+        $this->startId = $startId;
+        $this->endId = $endId;
+    }
+
+    /**
+     * Execute the job.
+     */
+    public function handle(): void
+    {
+        // update region code from FIPS to ISP, according to the entries in the table region_mapping_tmp
+        // Laravel join+update does not work well with PostgreSQL, so use the direct SQLs
+        // daily
+        if (substr(Config::getVar('database', 'driver'), 0, strlen('postgres')) === 'postgres') {
```

Comment:
I think it should be possible to build this update and the others using the Laravel query builder.